op-node/p2p/peer_scorer.go

```diff
@@ -1,16 +1,92 @@
 package p2p
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+
 	log "github.com/ethereum/go-ethereum/log"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	peer "github.com/libp2p/go-libp2p/core/peer"
 )
 
 type scorer struct {
-	peerStore Peerstore
-	metricer  GossipMetricer
-	log       log.Logger
-	gater     PeerGater
+	peerStore           Peerstore
+	metricer            GossipMetricer
+	log                 log.Logger
+	gater               PeerGater
+	bandScoreThresholds BandScorer
+}
+
+// bandScoreThresholds holds the thresholds for classifying peers
+// into different score bands.
+type bandScoreThresholds struct {
+	bands      map[string]float64
+	lowestBand string
+}
+
+// BandScorer is an interface for placing peer scores
+// into various bands.
+//
+// Implementations are expected to construct internals using the
+// [Parse] function and then expose the [Bucket] function for
+// downstream [BandScorer] consumers.
+//
+//go:generate mockery --name BandScorer --output mocks/
+type BandScorer interface {
+	Parse(str string) error
+	Bucket(score float64) string
+	Reset()
+}
+
+// NewBandScorer constructs a new [BandScorer] instance.
+func NewBandScorer() BandScorer {
+	return &bandScoreThresholds{
+		bands: make(map[string]float64),
+	}
+}
+
+// Reset wipes the internal state of the [BandScorer].
+func (s *bandScoreThresholds) Reset() {
+	s.bands = make(map[string]float64)
```

Comment:
It is more efficient in Go to reuse maps, especially here because they retain the same cardinality and keys. You might want to iterate here and set all values to 0. Or you're lazy and [`Clear`](https://pkg.go.dev/golang.org/x/exp/maps#Clear) it.